src/Compilers/CSharp/Test/Emit/CodeGen/PatternTests.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeGen;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.CodeGen
+{
+    public class PatternTests : EmitMetadataTestBase
```

Comment:
I would put these tests next to the other pattern-matching tests. It is very convenient when all tests for a feature are next to each other, rather than spread across different test binaries.